static/app/views/profiling/functions/index.tsx

```diff
@@ -0,0 +1,147 @@
+import {Fragment, useEffect, useMemo, useState} from 'react';
+import {browserHistory} from 'react-router';
+import * as Sentry from '@sentry/react';
+import {Location} from 'history';
+
+import {Client} from 'sentry/api';
+import * as Layout from 'sentry/components/layouts/thirds';
+import NoProjectMessage from 'sentry/components/noProjectMessage';
+import PageFiltersContainer from 'sentry/components/organizations/pageFilters/container';
+import SentryDocumentTitle from 'sentry/components/sentryDocumentTitle';
+import {t} from 'sentry/locale';
+import {Organization, PageFilters, Project, RequestState} from 'sentry/types';
+import {VersionedFunctionCalls} from 'sentry/types/profiling/core';
+import {defined} from 'sentry/utils';
+import {decodeScalar} from 'sentry/utils/queryString';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+import useProjects from 'sentry/utils/useProjects';
+import withPageFilters from 'sentry/utils/withPageFilters';
+
+import {ProfilingHeader} from '../header';
+import {generateProfilingRoute} from '../routes';
+
+import {FunctionsContent} from './content';
+
+interface Props {
+  location: Location;
+  params: {
+    projectId?: Project['slug'];
+  };
+  selection?: PageFilters;
+}
+
+function FunctionsPage(props: Props) {
+  const api = useApi();
+  const organization = useOrganization();
+  const {projects} = useProjects();
+
+  const version = decodeScalar(props.location.query.version);
+  const transaction = decodeScalar(props.location.query.transaction);
+
+  const [requestState, setRequestState] = useState<RequestState<VersionedFunctionCalls>>({
+    type: 'initial',
+  });
+
+  const project = useMemo(
+    () => projects.find(p => p.slug === props.params.projectId),
+    [projects, props.params.projectId]
+  );
+
+  useEffect(() => {
+    if (!defined(props.params.projectId) || !defined(transaction) || !defined(version)) {
+      // the functions page requires that the project, transaction, and version be defined
+      // if they aren't defined, we choose to redirect the user back to the profiling
+      // landing page
+      browserHistory.push(generateProfilingRoute({orgSlug: organization.slug}));
+      return undefined;
+    }
```

Comment:
Why not display an error here?

Reply:
Fair point, I was just copying patterns we had elsewhere in sentry. Will update this to show an error